zcash_primitives/src/transaction/fees/zip317.rs

```diff
@@ -193,3 +212,51 @@ impl super::FeeRule for FeeRule {
             .ok_or_else(|| BalanceError::Overflow.into())
     }
 }
+
+#[cfg(zcash_unstable = "nu7")]
+impl super::ZSAFeeRule for FeeRule {
+    fn fee_required_zsa<P: consensus::Parameters>(
+        &self,
+        _params: &P,
+        _target_height: BlockHeight,
+        transparent_input_sizes: impl IntoIterator<Item = transparent::InputSize>,
+        transparent_output_sizes: impl IntoIterator<Item = usize>,
+        sapling_input_count: usize,
+        sapling_output_count: usize,
+        orchard_action_count: usize,
+        asset_creation_count: usize,
+        issue_note_count: usize,
```

Comment:
unclear from name and doc if this is total notes or only the newly issued notes.